### ISSUE TITLE
Alarm triggered by Unavailable entity - while "Trigger when unavailable" is off

### ISSUE THREAD

**Issue by zbuh** [nielsfaber/alarmo#72]
1- This night the alarmo was triggered, by the motion sensor on garage.
I checked the logs, and the entity was unavailable by an instant.
The configuration: Trigger when unavailable = off
So I think we have a bug here.

2- In the same event, I also noticed something wrong with the notifications.
The message sent on that event (triggered - armed_night) was sent more than 20x. I think that it should only be sent once. Am I correct?

**Comment by Daider75** [comment]
Hello. I had a similar experience this weekend. My Alarm went to times off because of unavailable sensors.
Both times it was a different sensor.
I also have deactivated "Trigger when unavailable"
I have for now deactivated the main siren to not disturbing the neighbors.

Alarmo is really great so far. Thanks

**Comment by cokeman0** [comment]
Hi,

Also seen this today, devices from deconz, became unavailable.

```
2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.bevagelse_viktualie changed: old_state=<state binary_sensor.bevagelse_viktualie=off; on=True, friendly_name=Bevægelse Viktualie, device_class=motion @ 2020-12-20T17:22:24.622822+01:00>, new_state=<state binary_sensor.bevagelse_viktualie=unavailable; friendly_name=Bevægelse Viktualie, device_class=motion @ 2020-12-21T11:45:30.635608+01:00>

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.automations] state is updated from armed_away to pending

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.automations] executing automation 1607711884

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_4 changed: old_state=<state binary_sensor.presence_4=off; on=True, temperature=23.0, friendly_name=Motion Sensor - Mellem gang, device_class=motion @ 2020-12-20T17:22:24.623062+01:00>, new_state=<state binary_sensor.presence_4=unavailable; friendly_name=Motion Sensor - Mellem gang, device_class=motion @ 2020-12-21T11:45:30.635996+01:00>

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.motion_philips1 changed: old_state=<state binary_sensor.motion_philips1=off; on=True, friendly_name=Philips motion - mellem gang, device_class=motion @ 2020-12-20T17:22:24.623548+01:00>, new_state=<state binary_sensor.motion_philips1=unavailable; friendly_name=Philips motion - mellem gang, device_class=motion @ 2020-12-21T11:45:30.636753+01:00>

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_15 changed: old_state=<state binary_sensor.presence_15=off; on=True, temperature=23.0, friendly_name=Badeværelset, device_class=motion @ 2020-12-20T17:22:24.624067+01:00>, new_state=<state binary_sensor.presence_15=unavailable; friendly_name=Badeværelset, device_class=motion @ 2020-12-21T11:45:30.637593+01:00>

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.tradfri_motion_sensor changed: old_state=<state binary_sensor.tradfri_motion_sensor=off; on=True, friendly_name=TRÅDFRI Motion sensor WC, device_class=motion @ 2020-12-20T17:22:24.624319+01:00>, new_state=<state binary_sensor.tradfri_motion_sensor=unavailable; friendly_name=TRÅDFRI Motion sensor WC, device_class=motion @ 2020-12-21T11:45:30.637809+01:00>

2020-12-21 11:45:30 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.tradfri_motion_sensor_2 changed: old_state=<state binary_sensor.tradfri_motion_sensor_2=off; on=True, dark=True, friendly_name=TRÅDFRI Motion entre, device_class=motion @ 2020-12-21T11:38:56.858463+01:00>, new_state=<state binary_sensor.tradfri_motion_sensor_2=unavailable; friendly_name=TRÅDFRI Motion entre, device_class=motion @ 2020-12-21T11:45:30.638032+01:00>

2020-12-21 11:46:00 DEBUG (MainThread) [custom_components.alarmo.alarm_control_panel] async_entry_timer_finished

2020-12-21 11:46:00 DEBUG (MainThread) [custom_components.alarmo.automations] state is updated from pending to triggered
```

**Comment by nielsfaber** [comment]
This is concerning. I will investigate, hope I can reproduce this issue and improve the robustness.
I will keep you posted.

**Comment by nielsfaber** [comment]
Unfortunately I was not able to reproduce the issues that you experience.
However, I released v1.4.1 which comes with a rewritten implementation of the triggering behaviour of sensors.
There is a good change this solves the issues, so I would like to ask you to update and keep me posted of the experiences.

As for:
> 2- In the same event, I also noticed something wrong with the notifications.
The message sent on that event (triggered - armed_night) was sent more than 20x. I think that it should only be sent once. Am I correct?

This behaviour is not changed in the update, so it can occur again.
If you experience this again, please share the logging of this event.

**Comment by cokeman0** [comment]
Hi,

I have some devices (TRÅDFRI and other, this trigger is on a TRÅDFRI device) , connected via deconz, and when the deconz integration is reloaded, this happens. might not be related to manufacture. 

```
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.bevagelse_viktualie changed: old_state=closed, new_state=unknown
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] Alarm is triggered due to sensor: binary_sensor.bevagelse_viktualie
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.automations] state is updated from armed_away to pending
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.automations] executing automation 1607711884
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_4 changed: old_state=closed, new_state=unknown
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.motion_philips1 changed: old_state=closed, new_state=unknown
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_15 changed: old_state=closed, new_state=unknown
2020-12-23 13:22:58 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.tradfri_motion_sensor changed: old_state=closed, new_state=unknown
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.bevagelse_viktualie changed: old_state=unknown, new_state=unavailable
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] Entity binary_sensor.bevagelse_viktualie should not trigger on unavailable state, ignoring
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_4 changed: old_state=unknown, new_state=unavailable
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] Entity binary_sensor.presence_4 should not trigger on unavailable state, ignoring
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.motion_philips1 changed: old_state=unknown, new_state=unavailable
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] Entity binary_sensor.motion_philips1 should not trigger on unavailable state, ignoring
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_15 changed: old_state=unknown, new_state=unavailable
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] Entity binary_sensor.presence_15 should not trigger on unavailable state, ignoring
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.tradfri_motion_sensor changed: old_state=unknown, new_state=unavailable
2020-12-23 13:22:59 DEBUG (MainThread) [custom_components.alarmo.sensors] Entity binary_sensor.tradfri_motion_sensor should not trigger on unavailable state, ignoring
2020-12-23 13:23:00 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.bevagelse_viktualie changed: old_state=unavailable, new_state=closed
2020-12-23 13:23:00 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_4 changed: old_state=unavailable, new_state=closed
2020-12-23 13:23:00 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.motion_philips1 changed: old_state=unavailable, new_state=closed
2020-12-23 13:23:00 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.presence_15 changed: old_state=unavailable, new_state=closed
2020-12-23 13:23:00 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.tradfri_motion_sensor changed: old_state=unavailable, new_state=closed
2020-12-23 13:23:28 DEBUG (MainThread) [custom_components.alarmo.alarm_control_panel] async_entry_timer_finished
2020-12-23 13:23:28 DEBUG (MainThread) [custom_components.alarmo.automations] state is updated from pending to triggered
```

Triggered on "Alarm is triggered due to sensor: binary_sensor.bevagelse_viktualie" but the log also shows "binary_sensor.bevagelse_viktualie should not trigger on unavailable state, ignoring"

I can reproduce every time I reload deconz integration

**Comment by nielsfaber** [comment]
@cokeman0 first logging shows:
> entity binary_sensor.bevagelse_viktualie changed: old_state=closed, new_state=unknown

The state became `unknown`, which is not the same as `unavailable`. 
Ofcourse I could also consider to have this state to be ignored as well.
IMO this is a glitch in the Deconz integration, since the entities should become `unavailable` when the gateway goes offline.

**Comment by cokeman0** [comment]
You are right, but I can also trigge the alarm, by reloading the vera integration, showing almost the same in the logs. 

```
2020-12-23 14:09:18 DEBUG (MainThread) [custom_components.alarmo.sensors] entity binary_sensor.door_garden_523 changed: old_state=closed, new_state=unknown
2020-12-23 14:09:18 DEBUG (MainThread) [custom_components.alarmo.sensors] Alarm is triggered due to sensor: binary_sensor.door_garden_523
2020-12-23 14:09:18 DEBUG (MainThread) [custom_components.alarmo.automations] state is updated from armed_away to pending
etc etc etc :-)
```

So this issue, might be relates to all integrations devices...

**Comment by cokeman0** [comment]
And I have now testet openzwave integration (openzwave QT), and this "bug" also shows up there

**Comment by nielsfaber** [comment]
@cokeman0 Thank you for testing. 
I will make changes such that unavailable and unknown state will be treated the same.
Please give me a week or so for the new update (it also includes the new feature of multiple area support).

**Comment by Daider75** [comment]
> @cokeman0 Thank you for testing.
> I will make changes such that unavailable and unknown state will be treated the same.
> Please give me a week or so for the new update (it also includes the new feature of multiple area support).

Hello Niels, happy new year, I hope you started the new year in the best way.
Thanks for your hard work. Just for letting you know, I haven't had any false alarm the last 10 days. It happened one time more after last time I reported but since then everything looks fine. May be version 1.4.1 did the trick for me. Nevertheless I will keep my external siren off until the new version.
Thanks for your hard work on this.
Best regards
Dieter

**Comment by cokeman0** [comment]
The "NONE" bug has been fixed in ver 1.5...
